### ISSUE TITLE
Request: Filter via More than just Filename

### ISSUE THREAD

**Issue by MichaelTunnell** [robinmalburn/sublime-tabfilter#6]
I wanted this kind of filter feature because I often have a lot of the same types of files that are the same filename but are in different folders. For example, I have todo.txt files separated into folders for organization.

Structure:
- /todos/category-a/subcategory-a/todo.txt
- /todos/category-b/subcategory-e/todo.txt
- /todos/category-c/subcategory-d/todo.txt

The tab names become something like:
_todo.txt - category-b/subcategory-e_

I'd like to be able to filter via the folder differences in the name rather than the filenames.

Is this possible?

**Comment by robinmalburn** [comment]
Hmm, I think that should be doable.  I think the most sensible way to handle that in a "generic-works-for-everyone" sort of way would be to add a configuration that allows filtering the list by the full pathname - path and filename included.  You'd get the vanilla behaviour this way, and you'd get the behaviour you're looking for, too.  Does that work for your use case, do you think?

Realistically, I don't think it would be super hard to wire up that logic (famous last words).  The code around populating that popup at the moment is a bit cluttered and this would definitely make it far worse, so probably the most painful bit would be refactoring that section to be more readable and maintainable.  I switched back to vim a while back since ST3 was being an unstable mess at the time, but dusting off ST3 to work on a new feature for the plugin isn't a problem (always nice to a have a reason to write Python).

**Comment by MichaelTunnell** [comment]
> add a configuration that allows filtering the list by the full pathname - path and filename included. You'd get the vanilla behaviour this way, and you'd get the behaviour you're looking for, too. Does that work for your use case, do you think?

That sounds good to me. I don't mind changing a preference option for this. This would solve the biggest headache I have with multiple files of the same filename. Right now, I have to use a mouse and it is driving me crazy. :)

> I switched back to vim a while back since ST3 was being an unstable mess at the time, but dusting off ST3 to work on a new feature for the plugin isn't a problem 

That's super awesome of you, thanks!

**Comment by robinmalburn** [comment]
A new version of the plugin is up and should be available via Package Control now (though the Package Control site is still showing the old version on there, when I actually did the install I got the new version, so your mileage may vary).

This feature is off by default since it's probably not the standard use case, but you can turn it on by adding a new `include_path` option to the plugin's settings.  I think I've got the behaviour pretty much exactly as you described - the common path parts (well, common _text_ to be honest - but that was a concession I had to make to keep SublimeText 2 support and should largely go unnoticed the way I've implemented) will be trimmed off, so even with the feature turned on, you're only really going to see it when it would make sense (i.e. when you've got duplicate filenames across differing paths).  Honestly, I was half tempted to default the feature to on since it feels fairly unobtrusive, but it'd be bound to annoy someone if I did, so figured I'd walk the line and try and make everyone happy ;).

Hopefully this makes ticks all the boxes for you - let me know if there are any rough edges I've missed (or glaringly obvious pitfalls I've been totally oblivious too) and I'd be happy to try tinker a bit more with it.

**Comment by MichaelTunnell** [comment]
I just upgraded to the new version with this enhancement and it is absolutely perfect for my needs. I don't really have a preference as to whether this is enabled by default or not because having it in general is fantastic to me.

I just tested it with my todo setup and it performed exactly as I was hoping for.

Thank You VERY MUCH! This plugin is great overall and now it perfectly suits my needs and I couldn't be happier. :)
